public/src/main/java/com/revenuecat/purchases/models/GoogleProrationMode.kt

```diff
@@ -25,5 +30,64 @@ enum class GoogleProrationMode(@BillingFlowParams.ProrationMode val playBillingC
      * The purchase will fail if this mode is used when switching between [SubscriptionOption]s
      * of the same [StoreProduct].
      */
-    IMMEDIATE_WITH_TIME_PRORATION(BillingFlowParams.ProrationMode.IMMEDIATE_WITH_TIME_PRORATION)
+    IMMEDIATE_WITH_TIME_PRORATION(BillingFlowParams.ProrationMode.IMMEDIATE_WITH_TIME_PRORATION),
+
+    /**
+     * Replacement takes effect when the old plan expires, and the new price will be charged at the same time.
+     *
+     * Example: Samwise's Tier 1 subscription continues until it expires on April 30. On May 1st, the
+     * Tier 2 subscription takes effect, and Samwise is charged $36 for his new subscription tier.
+     */
+    DEFERRED(BillingFlowParams.ProrationMode.DEFERRED),
+
+    /**
+     * Replacement takes effect immediately, and the user is charged full price of new plan and is
+     * given a full billing cycle of subscription, plus remaining prorated time from the old plan.
+     *
+     * Example: Samwise's Tier 1 subscription is immediately ended. His Tier 2 subscription begins today and he is
+     * charged $36. Since he paid for a full month but used only half of it, half of a month's subscription ($1)
+     * is applied to his new subscription. Since that new subscription costs $36/year, he would get 1/36th of a year
+     * added on to his subscription period (~10 days). Therefore, Samwise's next charge would be 1 year and 10 days
+     * from today for $36. After that, he is charged $36 each year following.
+     */
+    IMMEDIATE_AND_CHARGE_FULL_PRICE(BillingFlowParams.ProrationMode.IMMEDIATE_AND_CHARGE_FULL_PRICE),
+
+    /**
+     * Replacement takes effect immediately, and the billing cycle remains the same.
+     *
+     * Example: This mode can be used because the Tier 2 subscription price per time unit ($36/year = $3/month) is
+     * greater than Tier 1 subscription price per time unit ($2/month). Samwise's Tier 1 subscription is immediately
+     * ended.
+     * Since he paid for a full month but used only half of it, half of a month's subscription ($1) is applied to
+     * his new subscription. However, since that new subscription costs $36/year, the remaining 15 days costs $1.50,
+     * so he is charged the difference of $0.50 for his new subscription.
+     * On May 1st, Samwise is charged $36 for his new subscription tier and another $36 on May 1 of each year following.
+     */
+    IMMEDIATE_AND_CHARGE_PRORATED_PRICE(BillingFlowParams.ProrationMode.IMMEDIATE_AND_CHARGE_PRORATED_PRICE);
```

Comment:
So I might be wrong but I think we didn't add these before because they weren't fully supported? (I might be wrong) Just want to make sure we want to allow users to use these.

Reply:
We're working on supporting them in the BE